src/abstractdataframe/selection.jl

```diff
@@ -537,7 +541,14 @@ julia> combine(df, :a => sum, nrow)
 combine(df::AbstractDataFrame, args...) =
     manipulate(df, args..., copycols=true, keeprows=false)
 
-combine(arg, df::AbstractDataFrame) = combine(arg, groupby(df, []))
+function combine(arg, df::AbstractDataFrame)
+    if nrow(df) == 0
+        throw(ArgumentError("calling combine on a data frame with zero rows" *
+                            " with transformation as a first argument is " *
+                            "currently not supported"))
+    end
+    return combine(arg, groupby(df, []))
```

Comment:
While we're at it, maybe this will avoid compiling a method for `Any[]` which should be rarely used? I wonder this pattern is used elsewhere.
```suggestion
    return combine(arg, groupby(df, Symbol[]))
```